code/datums/components/slippery.dm

```diff
@@ -0,0 +1,22 @@
+/datum/component/slippery
+	var/intensity
+	var/lube_flags
+	var/mob/slip_victim
+
+/datum/component/slippery/New(datum/P, _intensity, _lube_flags = NONE)
+	..()
+	intensity = max(_intensity, 0)
+	lube_flags = _lube_flags
+	if(ismovableatom(P))
+		RegisterSignal(COMSIG_MOVABLE_CROSSED, .proc/Slip)
+	else
+		RegisterSignal(COMSIG_ATOM_ENTERED, .proc/Slip)

```

Comment:
Technically you can have movables that slip on entered but it's not a thing right now so this is fine.